vyper/builtins/functions.py

```diff
@@ -294,6 +294,46 @@ class Slice(BuiltinFunction):
     ]
     _return_type = None
 
+    def evaluate(self, node):
+        validate_call_args(node, 3)
+        literal_value, start, length = node.args
+        if (
+            isinstance(literal_value, (vy_ast.Bytes, vy_ast.Str, vy_ast.Hex))
+            and isinstance(start, vy_ast.Int)
+            and isinstance(length, vy_ast.Int)
+        ):
+            (start_val, length_val) = (start.value, length.value)
+
+            if start_val < 0:
+                raise ArgumentException("Start cannot be negative", start)
+            elif length_val <= 0:
+                raise ArgumentException("Length cannot be negative", length)
+
+            if isinstance(literal_value, vy_ast.Hex):
+                if start_val >= 32:
+                    raise ArgumentException("Start cannot take that value", start)
+                if length_val > 32:
+                    raise ArgumentException("Length cannot take that value", length)
+                length = len(literal_value.value) // 2 - 1
+                if length != 32:
+                    raise ArgumentException("Length can only be of 32", literal_value)
+                start_val *= 2
+                length_val *= 2
+
+            if start_val + length_val > len(literal_value.value):
+                raise ArgumentException("Slice is out of bounds", start)
+
+            if isinstance(literal_value, vy_ast.Bytes):
+                sublit = literal_value.value[start_val : (start_val + length_val)]
```

Comment:
it's going to be easier if we have a helper variable like `end = start + length`

Reply:
nice one thanks